cluster/src/dunit/scala/io/snappydata/cluster/SecureQueryRoutingDUnitTest.scala

```diff
@@ -0,0 +1,246 @@
+    builder.append(s"$qryTest Stmt: Total number of rows = $index").append("\n")
+    // scalastyle:off println
+    println(builder.toString())
+    // scalastyle:on println
+    assert(index == 40000)
+  }
+
+  def query1(serverHostPort: Int, tableName: String, user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"query1: Connected to $serverHostPort")
+    // scalastyle:off println
+
+    val stmt1 = conn.createStatement()
+    try {
+      val qry1 = s"select ol_int_id, ol_int2_id, ol_str_id " +
+          s" from $tableName " +
+          s" where ol_int_id < 5000000 " +
+          s""
+      val rs1 = stmt1.executeQuery(qry1)
+      verifyQuery1(qry1, rs1)
+      rs1.close()
+      // Thread.sleep(1000000)
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def createTable1(serverHostPort: Int, tableName: String, user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"createTable1: Connected to $serverHostPort")
+    // scalastyle:on println
+
+    val stmt1 = conn.createStatement()
+    try {
+    stmt1.execute(s"create table $tableName (ol_int_id  integer," +
+        s" ol_int2_id  integer, ol_str_id STRING) using column " +
+        "options( partition_by 'ol_int_id, ol_int2_id', buckets '5', COLUMN_BATCH_SIZE '200')")
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def dropTable(serverHostPort: Int, tableName: String, user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"dropTable1: Connected to $serverHostPort")
+    // scalastyle:on println
+
+    val stmt1 = conn.createStatement()
+    try {
+      stmt1.execute(s"drop table $tableName")
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def createTable2(serverHostPort: Int, tableName: String, user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"createTable2: Connected to $serverHostPort")
+    // scalastyle:on println
+
+    val stmt1 = conn.createStatement()
+    try {
+      stmt1.execute(s"create table $tableName (ol_int_id  integer," +
+          s" ol_int2_id  integer, ol_str_id STRING) using row " +
+          "options( partition_by 'ol_int_id, ol_int2_id', buckets '5')")
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+  
+  def testColumnTableRouting(): Unit = {
+    val serverHostPort = AvailablePortHelper.getRandomAvailableTCPPort
+    vm2.invoke(classOf[ClusterManagerTestBase], "startNetServer", serverHostPort)
+    // scalastyle:off println
+    println(s"network server started at $serverHostPort")
+    // scalastyle:on println
+
+    val tableName = "order_line_col"
+    createTable1(serverHostPort, tableName, jdbcUser, jdbcUser)
```

Comment:
Where is the negative testing. Like a user who is not authorized to touch tabl1 created by jdbcUser